### ISSUE TITLE
issues need resolving.

### ISSUE THREAD

**Issue by zeffii** [nortikin/sverchok#3159]
i'm going to say a few unpopular things here. 

The update system in the current master is broken, the bug may be easy to fix, but it's there and i will not be the only person experiencing it. I think it was ill-advised to full on merge various update system augmentations directly into master without first having them merged into a dedicated `nortikin/some_branch` test for more convenient testing. The allure of a snappier update system was too powerful : )

as @ly29 suggested a while back, the existing update system has worked reasonably well for a long time now with only minor modifications at any given moment.

so.. what to do..?
- fix that bug ( those bugs? ...)
- how long to keep master in this state ?
- is it a waste of time pushing more commits onto master in this state
- do we revert to https://github.com/nortikin/sverchok/commit/476f26dee0f3faf5ebfea52464d4b241e3a1ae0a last known good configuration , or https://github.com/nortikin/sverchok/pull/3046

**Comment by zeffii** [comment]
how persistent are 
```
class SvNodeTreeCommon(object):
    ...
    sv_links = SvLinks()
    nodes_dict = SvNodesDict()
```
between reloads?

**Comment by Durman** [comment]
I think reverting of next additions from master can bring it to stable condition especially last one.
#3134
#3089
#3046

**Comment by zeffii** [comment]
yeah #3046

**Comment by vicdoval** [comment]
I would not say the update system is broken, is the reload event what is broken.
Have you experience any issue during a regular sverchok session? 
I understand this is specially annoying for us, becacuse the regular user does no reload scripts ever. Just opens and closes files.
After #3046 when I have been using Sverchok as user i have been delighted with its performace.
The problem of the branches is the we don't have testers, so none of this issues appear when it was in a branch.
Let me study it today and if i don't find a solution I will turn the last updates into a "experimental feature" that can be activated through the preferences

**Comment by ly29** [comment]
On reload, reset everything basically is the only sensible strategy.

**Comment by zeffii** [comment]
yeah, the system seems to behave well until i refresh python. Makes it painful to debug my own branches which may themselves not be entirely stable

**Comment by portnov** [comment]
For me, sometimes the node tree does not get recalculated when I link / unlink nodes. Did anyone notice something like that?

**Comment by zeffii** [comment]
yes @portnov ,  come to think of it that has happened too. sometimes i can retrigger an update with a framechange (left right) or updating a node upstream from the "stale" node

**Comment by vicdoval** [comment]
I think now it is fixed please test and report 😃

**Comment by zeffii** [comment]
this seems to have fixed the most bothersome issue for me. Little use to keep this issue open.

maybe in the future we should consider a few steps for larger commits that affect
   - the update system in particular  or
   - a large portion of existing nodes.

1.  merge it into `nortikin/master_development_featurename`
2.  get a few others to actively test it for for sanity, because travis will not always find everything.
